### PR TITLE
Console event subscribing

### PR DESCRIPTION
**Pull request by luceos**
**Relates to #1789**

> Please note this PR does not add error logging for console in core!

**Changes proposed in this pull request:**

This PR adds an EventDispatcher to the Console Application. This event dispatcher is passed on to the Configuring event so that extensions can hook into it.

**Reviewers should focus on:**

This is one path for implementation. It's the cleanest I could think of. I think we don't want or need a bloated console side of our software. This implementation adds a dependency to the symfony/event-dispatcher package, this package is required to match the Interface requirement inside Symfony Application.

An alternative implementation would be to start using Illuminate Console instead. The implementation would possibly be easier, no research done yet.

An example implementation of this proposal can be found on the fof/sentry package, PR: https://github.com/FriendsOfFlarum/sentry/pull/1

- [ ] ~~Frontend changes: tested on a local Flarum installation.~~
- [ ] Backend changes: tests are green (run `php vendor/bin/phpunit`).

### CODE REVIEW

**Comment by luceos** [comment]
Should this PR be based off of your new exception handling logic @franzliedke ?

**Comment by franzliedke** [comment]
Yeah, I was hoping this would be possible now.

**Comment by luceos** [comment]
In order to add your new Error Registry and handler I need to merge this to move forward, can you review this approach @franzliedke ?

**Comment by luceos** [comment]
I am merging this as I see no alternative right now. Let me know if you disagree.

File: composer.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
         "s9e/text-formatter": "^1.2.0",
         "symfony/config": "^3.3",
         "symfony/console": "^4.2",
+        "symfony/event-dispatcher": "^4.3.2",
```
Comment (franzliedke) [inline composer.json]: Why the specific constraint?

File: src/Console/Server.php [hunk 2]
```diff
@@ -46,6 +47,8 @@ private function extend(ConsoleApplication $console)
         $app = Application::getInstance();
 
         $events = $app->make(Dispatcher::class);
-        $events->fire(new Configuring($app, $console));
+        $events->fire(new Configuring($app, $console, $dispatcher = new EventDispatcher()));
+
+        $console->setDispatcher($dispatcher);
```
Comment (franzliedke) [inline src/Console/Server.php]: I was assuming we would handle events ourselves and pass the exceptions to our array of [reporters registered with the application](https://github.com/flarum/core/blob/12dc4fff573b7cc053e6eaf95756297469cfb7b0/src/Foundation/ErrorServiceProvider.php#L73). That way, the Sentry exception would not need any further changes.

If you would furthermore pipe the error through the registry, [you could use that](https://github.com/flarum/core/blob/12dc4fff573b7cc053e6eaf95756297469cfb7b0/src/Http/Middleware/HandleErrors.php#L61-L67) to determine whether the error is even worth reporting.

**Pull request merged**